### PR TITLE
Resolve Upper Bound issues, update Groovy

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] - Update parent POM
- [x] - Add upper bound enforcement
- [x] - cleanup issues (mostly commons-*)
- [x] - Update Groovy to 2.4.11, so it is the same as the core baseline

All versions are compatible with the Jenkins core.

@reviewbybees @jglick

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -44,12 +44,22 @@
     <dependency>
       <groupId>commons-beanutils</groupId>
       <artifactId>commons-beanutils</artifactId>
-      <version>1.7.0</version>
+      <version>1.8.0</version>
```
Comment (jglick) [inline core/pom.xml]: Are these aligned with `jenkins-core`?

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.kohsuke</groupId>
     <artifactId>pom</artifactId>
-    <version>14</version>
+    <version>19</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Ideally we should start migrating it to https://github.com/jenkinsci/pom.
IIUC there is no other things excepting Plugin site release and code signing there. WDYT @jglick ?

**Comment by jglick** [comment]
Oops:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project stapler-parent: failed to get report for org.apache.maven.plugins:maven-javadoc-plugin: Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M1:enforce (default) on project stapler: Execution default of goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M1:enforce failed. NullPointerException -> [Help 1]
```

File: core/pom.xml (hunk 1 above)
Comment (oleg-nenashev) [reply core/pom.xml]: Jenkins core is at 1.8.3. Can bump it even more

**Comment by oleg-nenashev** [comment]
Works on my machine (c). No sure what happens there

File: pom.xml (hunk 2 above)
Comment (jglick) [reply pom.xml]: Not sure. Technically this is a non-Jenkins component.

Comment (oleg-nenashev) [reply pom.xml]: It's something we could also fix. Stapler is a critical part of Jenkins && it's not extremely  popular elsewhere, so moving it && integrating common Jenkins QA/review flow would not hurt. It's something we were discussing multiple times, but maybe it belongs to the ML now.

**Comment by jglick** [comment]
Reproducible locally:

```
java.lang.NullPointerException
	at org.apache.maven.plugins.enforcer.EnforceBytecodeVersion.isBadArtifact(EnforceBytecodeVersion.java:195)
	at org.apache.maven.plugins.enforcer.EnforceBytecodeVersion.checkDependencies(EnforceBytecodeVersion.java:182)
	at org.apache.maven.plugins.enforcer.EnforceBytecodeVersion.handleArtifacts(EnforceBytecodeVersion.java:109)
	at org.apache.maven.plugins.enforcer.AbstractResolveDependencies.execute(AbstractResolveDependencies.java:77)
	at org.apache.maven.plugins.enforcer.EnforceMojo.execute(EnforceMojo.java:202)
```

You just need to update `extra-enforcer-rules` to `1.0-beta-4` or later to pick up the fix of https://github.com/mojohaus/extra-enforcer-rules/issues/13.

**Comment by oleg-nenashev** [comment]
@jglick ptal

File: pom.xml (hunk 2 above)
Comment (jglick) [reply pom.xml]: Basically that would mean not releasing it to Central any more. Fine with me, but needs some discussion.

**Comment by jglick** [comment]
It seems `NumberConverter` regressed sometime between 1.7.0 and 1.8.0. 1.8.3 is also affected. But 1.9.3 is even worse—it does not convert `String[]`s properly either. Any I think I have identified a bug in how Stapler was calling BeanUtils that is responsible. Will file a follow-up PR.

**Pull request merged**